File: templates/csharp12/Starter.cs
using System;

/*
Starter template for C# 12 (.NET 8.0)
Entrypoint: Run with `dotnet run` (requires a .csproj) or compile with `csc Starter.cs` and run `./Starter`
For simpler testing: `dotnet script Starter.cs` (requires dotnet-script tool)
Sample I/O:
  Input: (via stdin or args)
  Output: (via stdout)
*/

class Starter
{
    static void Main(string[] args)
    {
        // Read input
        Console.Write("Enter your name: ");
        string? input = Console.ReadLine();

        // Process
        string result = $"Hello, {input ?? "World"}!";

        // Write output
        Console.WriteLine(result);
    }
}

File: templates/java25/Starter.java
import java.util.Scanner;

/**
 * Starter template for Java 25
 * Entrypoint: Compile with `javac Starter.java` and run with `java Starter`
 * Sample I/O:
 *   Input: (via stdin or args)
 *   Output: (via stdout)
 */
public class Starter {
    public static void main(String[] args) {
        // Read input
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter your name: ");
        String input = scanner.nextLine();
        scanner.close();

        // Process
        String result = String.format("Hello, %s!", input);

        // Write output
        System.out.println(result);
    }
}
